### ISSUE TITLE
vflip fliter throws an exception on FFmpegFrameFilter

### ISSUE THREAD

**Issue by SIY1121** [bytedeco/javacv#975]
Hi
I have one bug report and question.

I'm using javacv as FFmpeg wrapper.
I received raw video frames and tried to flip frames when encoding, but using vflip fliter throws an exception.

```Java
java.lang.IllegalArgumentException
	at java.nio.Buffer.limit(Buffer.java:275)
	at org.bytedeco.javacv.FFmpegFrameFilter.pull(FFmpegFrameFilter.java:359)
```

On debugging,I identified the cause.
With vflip filter, the imageStride of the frame returned by ffmpeg will be negative.

https://github.com/bytedeco/javacv/blob/51c5714faa487584c88be7dee8743e20cc3bc5f1/src/main/java/org/bytedeco/javacv/FFmpegFrameFilter.java#L582-L588

This process doesn't seem to consider it.
So I changed code to use absolute value and it works fine.
```Java
if (ptr != null && !ptr.equals(image_ptr[0])) { 
     image_ptr[0] = ptr.capacity(frame.imageHeight * Math.abs(frame.imageStride)); 
     image_buf[0] = ptr.asBuffer(); 
 } 
 frame.image = image_buf; 
 frame.image[0].position(0).limit(frame.imageHeight *  Math.abs(frame.imageStride)); 
 frame.imageChannels = frame.imageStride / Math.abs(frame.imageWidth);
```

I didn't use OpenCV flip function because process of converting array to mat and converting it to FFmpeg frame looks inefficiency.
Was it good choice?

Thank you.

**Comment by saudet** [comment]
Hum, interesting. Sounds like a bug in FFmpeg, but anyway, could you send a
pull request to fix this in JavaCV? Thanks!

**Comment by saudet** [comment]
Fixed with pull #977, thanks!

**Comment by saudet** [comment]
Which has now been integrated in version 1.4.2. Thanks again for the fix!
